pkg/synthetictests/platformidentification/types.go

```diff
@@ -40,6 +55,70 @@ func CloneJobType(in JobType) JobType {
 	}
 }
 
+func BuildJobVariants(ctx context.Context, clientConfig *rest.Config) JobVariants {
+
+	// we could log the error if there is value
+	jobType, _ := GetJobType(ctx, clientConfig)
+
+	jobVariants := JobVariants{}
+
+	if jobType != nil {
+		jobVariants.Topology = jobType.Topology
+		jobVariants.Release = jobType.Release
+		jobVariants.FromRelease = jobType.FromRelease
+		jobVariants.Network = jobType.Network
+		jobVariants.Platform = jobType.Platform
+		jobVariants.Architecture = jobType.Architecture
+	}
+
+	// add in other data like region, etc.
+	configClient, err := configclient.NewForConfig(clientConfig)
+	if err != nil {
+		return jobVariants
+	}
+
+	network, err := configClient.Networks().Get(ctx, "cluster", metav1.GetOptions{})
+	if err == nil && len(network.Spec.ClusterNetwork) > 0 {
+		if strings.Contains(network.Spec.ClusterNetwork[0].CIDR, ":") {
+			jobVariants.NetworkStack = "IPv6"
+		} else {
+			jobVariants.NetworkStack = "IPv4"
+		}
+	}
+
+	clusterVersions, err := configClient.ClusterVersions().List(ctx, metav1.ListOptions{})
+	if err == nil && clusterVersions != nil {
+		jobVariants.ClusterVersionHistory = getClusterVersions(clusterVersions)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return jobVariants
+	}
+
+	kNodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err == nil && kNodes != nil && len(kNodes.Items) > 0 {
+		jobVariants.CloudRegion = kNodes.Items[0].Labels[`topology.kubernetes.io/region`]
+		jobVariants.CloudZone = kNodes.Items[0].Labels[`topology.kubernetes.io/zone`]
+	}
+
+	return jobVariants
+}
+
+func getClusterVersions(versions *configv1.ClusterVersionList) []string {
+	if versions == nil {
+		return nil
+	}
+	cvs := make([]string, 0)
+	for _, v := range versions.Items {
+
+		for _, vv := range v.Status.History {
+			cvs = append(cvs, vv.Version)
+		}
+	}
+	return cvs
+}
```

Comment:
@deepsm007 / @stbenjam - I'm interested in your feedback on what this code produces

  [Sample](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade/1626951434970861568/artifacts/e2e-aws-ovn-upgrade/openshift-e2e-test/artifacts/junit/test-failures-summary_20230218-153052.json)

`
"ClusterVersionHistory": [
            "4.13.0-0.ci.test-2023-02-18-143051-ci-op-wk9qrjvn-initial",
            "4.13.0-0.ci.test-2023-02-18-144406-ci-op-wk9qrjvn-latest",
            "4.13.0-0.ci.test-2023-02-18-143051-ci-op-wk9qrjvn-initial"
        ]`
        
I'm presuming there are two entries in the versions.Items list but not sure.  Would the duplicate entry for 

`4.13.0-0.ci.test-2023-02-18-143051-ci-op-wk9qrjvn-initial`
    
throw things off?  Should I try limiting to just the first item in versions.Items and build the list from v.Status.History?

Reply:
Maybe we can add a uniqueItem func to avoid skipping other version.
There can be more than 2 version [4.13.0, 4.12.6,4.12.6,4.12.0] where 2 & 3 are same but 4th is different. If we try to limit to just first item we will miss that.

Reply:
Thanks @deep007 latest produces 
[job-variants](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade/1627729525368426496/artifacts/e2e-aws-ovn-upgrade/openshift-e2e-test/artifacts/junit/job-variants_20230220-185432.json)

` "ClusterVersionHistory": [
        "4.13.0-0.ci.test-2023-02-20-180102-ci-op-qhi2dvj0-initial",
        "4.13.0-0.ci.test-2023-02-20-181304-ci-op-qhi2dvj0-latest"
    ]`

Reply:
Lets run 1 rehearsal on metal to capture different platform

Reply:
I can look into metal.  We have the following & that agnostic is the one we are really chasing here so that is nice:

[vsphere](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-vsphere-ovn-etcd-scaling/1627729536000987136/artifacts/e2e-vsphere-ovn-etcd-scaling/openshift-e2e-test/artifacts/junit/job-variants_20230220-185821.json)

[agnostic](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-agnostic-ovn-cmd/1627729525108379648/artifacts/e2e-agnostic-ovn-cmd/openshift-e2e-test/artifacts/junit/job-variants_20230220-191006.json)

[gcp](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-gcp-ovn-etcd-scaling/1627729530984599552/artifacts/e2e-gcp-ovn-etcd-scaling/openshift-e2e-test/artifacts/junit/job-variants_20230220-185208.json)

[azure](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-azure-ovn-etcd-scaling/1627729525943046144/artifacts/e2e-azure-ovn-etcd-scaling/openshift-e2e-test/artifacts/junit/job-variants_20230220-191525.json)

[aws](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27731/pull-ci-openshift-origin-master-e2e-aws-ovn-etcd-scaling/1627729525192265728/artifacts/e2e-aws-ovn-etcd-scaling/openshift-e2e-test/artifacts/junit/job-variants_20230220-200705.json)

Reply:
[metal](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/openshift-origin-27731-nightly-4.13-e2e-metal-ipi-ovn-ipv6/1628038847461855232/artifacts/e2e-metal-ipi-ovn-ipv6/baremetalds-e2e-test/artifacts/junit/job-variants_20230221-162447.json)